cmake/modules/FindHOST_TOOLS.cmake

```diff
@@ -62,22 +62,24 @@ find_program(BOSSAC bossac)
 find_program(IMGTOOL imgtool)
 
 # Pick host system's toolchain if we are targeting posix
+zephyr_get(ZEPHYR_TOOLCHAIN_VARIANT)
 if("${ARCH}" STREQUAL "posix")
   if(NOT "${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "llvm")
-    set(ZEPHYR_TOOLCHAIN_VARIANT "host")
+    set(ZEPHYR_TOOLCHAIN_VARIANT "host" CACHE STRING "Zephyr toolchain variant" FORCE)
   endif()
 endif()
 
 # Keep XCC_USE_CLANG behaviour for a while.
 if ("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc"
     AND "$ENV{XCC_USE_CLANG}" STREQUAL "1")
```

Comment:
should `XCC_USE_CLANG` also use `zephyr_get`?

Reply:
no, because it's deprecated so we want to remove it completely.
Just haven't been deprecated long enough.

btw. this commit should be reviewed here #49672, 
From PR description:
> Note: the first 4 commits are cleanup commits that will be reviewed here: https://github.com/zephyrproject-rtos/zephyr/pull/49672 so ignore those commits when reviewing.
> Only part of this PR, as they are needed for zephyr_get() implementation to work properly.